src/adapter/src/coord/timeline.rs

```diff
@@ -337,6 +346,18 @@ impl<T: TimestampManipulation> DurableTimestampOracle<T> {
     }
 }
 
+/// An "object" that currently, or used to, exist on a [`Timeline`]
+///
+/// Before running a catalog transaction we'll check if any timelines would become empty
+/// after dropping resources, if so, we'll drop that timeline. But we only remove the
+/// timeline and/or the objects that are on it, after the transaction was successful.
+#[derive(Debug)]
+pub enum TimelineAssociation {
+    StorageId(GlobalId),
+    ComputeId((ComputeInstanceId, GlobalId)),
+    ComputeInstance(ComputeInstanceId),
+}
+
```

Comment:
We have an existing struct called `CollectionIdBundle` that mostly handles the same use case as this. The other methods in this file use `CollectionIdBundle`s and I think we can probably replace usage of this enum with that struct.

Reply:
Good call

Joe and I talked offline and settled on an approach using `CollectionIdBundle` and the preexisting `partition_ids_by_timeline_context` function which I had previously missed